packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -40,158 +44,304 @@ def ensure_routable(cmd, host, port):
+            app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                         '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+            app['portDefinitions'] = [{
+                'protocol': 'tcp',
+                'port': 0
+            }]
+            if vip is not None:
+                app['portDefinitions'][0]['labels'] = {'VIP_0': vip}
+            app['healthChecks'][0]['portIndex'] = 0
+
+        app['constraints'] = [['hostname', 'CLUSTER', host]]
+        log.info('app: {}'.format(json.dumps(app)))
+        return app, uuid
+
+    def vip_app(self, num, container, network, host, vip):
+        if container == 'UCR':
+            return self.mesos_vip_app(num, network, host, vip, ucr=True)
+        if container == 'DOCKER':
+            return self.docker_vip_app(network, host, vip)
+        if container == 'NONE':
+            return self.mesos_vip_app(num, network, host, vip, ucr=False)
+        assert False, 'unknown container option {}'.format(container)
+
+    def run(self, dcos_api_session):
+        self.log('START')
+        agents = list(copy.copy(dcos_api_session.all_slaves))
+        self.log('seed is {}'.format(hash(self.vip)))
+        random.seed(hash(self.vip))
+        random.shuffle(agents)
+        host1 = agents[0]
+        host2 = agents[0]
+        if not self.samehost:
+            host2 = agents[1]
+        log.debug('host1 is is: {}'.format(host1))
+        log.debug('host2 is is: {}'.format(host2))
+
+        origin_app, app_uuid = self.vip_app(self.num, self.container, self.vipnet, host1, self.vip)
+        origin_app['acceptedResourceRoles'] = ['*', 'slave_public']
+        proxy_app, _ = self.vip_app(self.num, self.container, self.proxynet, host2, None)
+        proxy_app['acceptedResourceRoles'] = ['*', 'slave_public']
+
+        returned_uuid = None
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app, timeout=timeout))
+            sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app, timeout=timeout))
+            proxy_host = sp[0].host
+            proxy_port = sp[0].port
+            if proxy_port == 0 and sp[0].ip is not None:
+                proxy_port = backend_port_st + self.num
+                proxy_host = sp[0].ip
+            log.info('proxy endpoints are {}'.format(sp))
+            cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(self.vipaddr)
+            returned_uuid = ensure_routable(cmd, proxy_host, proxy_port)
+            log.debug('returned_uuid is: {}'.format(returned_uuid))
+        assert returned_uuid is not None
+        assert returned_uuid['test_uuid'] == app_uuid
+        self.log('PASSED')
+
+
+class ProxyPod(Enum):
+    # NoPod = 0
+    ItsOwnPod = 1
+    SamePodAsVip = 2
+
+
+class PodTest():
+    def __init__(self, num, proxy_pod, proxy_net, vip_net, is_named, notes=''):
```

Comment:
can we please type these? We pass complex objects a lot in these tests and it would be helpful to instantly know these arguments are all str or whatever